backend/app/workers/llm.py

```
@@ -28,17 +30,37 @@
-def _resolve_request_id(model: LLMTaskPayload) -> str:
-    return model.request_id or str(uuid4())
+def _stable_fallback_id(model: LLMTaskPayload, endpoint: str, label: str) -> str:
+    payload = {
+        "tenant_id": str(model.tenant_id),
+        "endpoint": endpoint,
+        "correlation_id": model.correlation_id,
+        "request_id": model.request_id,
+        "prompt": model.prompt,
+    }
+    seed = json.dumps(payload, sort_keys=True, default=str)
+    digest = hashlib.sha256(f"{label}:{seed}".encode("utf-8")).hexdigest()
```

Ruff (0.14.11)

Unnecessary UTF-8 `encoding` argument to `encode`

Remove unnecessary `encoding` argument

(UP012)